com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/LambdaExpressionLocator.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+* Copyright (c) 2022 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Gayan Perera (gayanper@gmail.com) - initial API and implementation
+*******************************************************************************/
+package com.microsoft.java.debug;
+
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ASTVisitor;
+import org.eclipse.jdt.core.dom.AbstractTypeDeclaration;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.LambdaExpression;
+
+public class LambdaExpressionLocator extends ASTVisitor {
+    private CompilationUnit compilationUnit;
+    private int line;
+    private int column;
+    private boolean found;
+
+    private IMethodBinding lambdaMethodBinding;
+    private LambdaExpression lambdaExpression;
+
+    public LambdaExpressionLocator(CompilationUnit compilationUnit, int line, int column) {
+        this.compilationUnit = compilationUnit;
+        this.line = line;
+        this.column = column;
+    }
+
+    @Override
+    public boolean visit(LambdaExpression node) {
+        if (column > -1) {
+            int startPosition = node.getStartPosition();
+
+            int startColumn = this.compilationUnit.getColumnNumber(startPosition);
+            int endColumn = this.compilationUnit.getColumnNumber(startPosition + node.getLength());
+            int lineNumber = this.compilationUnit.getLineNumber(startPosition);
+
+            if (column >= startColumn && startColumn <= endColumn && lineNumber == line) {
```

Comment:
The algorithm is not accurate.

For the snippet `list.stream().map(name -> name).forEach((▮item) -> System.out.println(item))`, it always matches the first lambda `name -> name`.

Reply:
I think the issue is with line number.  I will fix it today evening and test the scenarios.

Reply:
I was able to fix the multi line issue and I think i can commit the both fixes including multiple lambda on same line by end of tomorrow

Reply:
i have now fixed and tested for both multiple lambda's on same line and lambdas spanning multiple lines.